applications/openshift/networking/default_ingress_ca_replaced/rule.yml

```diff
@@ -0,0 +1,47 @@
+prodtype: ocp4
+
+title: 'Ensure that the default Ingress CA (wildcard issuer) has been replaced'
+
+description: |-
+  Check that the default Ingress CA has been replaced.
+
+rationale: |-
+  OpenShift auto-generates several PKIs to serve TLS on different
+  endpoints of the system. It is possible and necessary to configure a
+  custom PKI which allows external clients to trust the endpoints.
+
+  The Ingress Operator is the component responsible for enabling external
+  access to OpenShift Container Platform cluster services. The aforementioned
+  operator creates an internal CA and issues a wildcard certificate that is
+  valid for applications under the .apps sub-domain. Both the web console
+  and CLI use this certificate as well. The certificate and key would need
+  to be replaced since a certificate coming from a trusted provider is
+  needed.
+
+  <tt>https://docs.openshift.com/container-platform/latest/security/certificates/replacing-default-ingress-certificate.html</tt>
+
+severity: medium
+
+references:
+  nist: SC-17
+
+ocil_clause: 'Ingress Operator default CA needs to be replaced'
+
+ocil: |-
+  Verify that the proxy/cluster resource contains the name of a trusted CA secret:
+    <tt>oc get proxy/cluster -ojsonpath='{ .spec."trustedCA"."name" }'</tt>
+  The result should be the name of a valid secret. If none, the default Ingress CA is in use.
+
+warnings:
+- general: |-
+    {{{ openshift_cluster_setting("/apis/config.openshift.io/v1/proxies/cluster") | indent(4) }}}
+
+template:
+  name: yamlfile_value
+  vars:
+    ocp_data: "true"
+    filepath: /apis/config.openshift.io/v1/proxies/cluster
+    yamlpath: ".spec.trustedCA.name"
```

Comment:
OpenSCAP's yaml probe has a caveat that for it to actually recognize that a rule's yamlpath doesn't match, the full path has to exist on the object. Now, `.spec.trustedCA.name` doesn't exist on the `cluster` object by default, which is why this reports PASS (confusingly enough).

In order to address this, we have the `check_existence_yamlpath` field which is documented here: https://complianceascode.readthedocs.io/en/latest/manual/developer/06_contributing_with_content.html#yamlfile-value

That field's purpose is to determine what's the minimum amount of yaml objects to gather... and so... if we compare it to what's being gathered they have to match. You can safely use `.metadata.name` which is always present in kubernetes objects. hence, in this case, `.metadata.name` would gather 1 object while `.spec.trustedCA.name` would gather 0 objects, and so it would fail which is what we want.

Reply:
Added, thanks!